lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -84,9 +84,7 @@ function gutenberg_get_global_stylesheet( $types = array() ) {
 	}
 	$tree                = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
 	$supports_theme_json = WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
-	if ( empty( $types ) && ! $supports_theme_json ) {
```

Comment:
I've noticed we've already done this for classic themes here (6.0 code): https://github.com/WordPress/gutenberg/pull/42012 It wasn't [backported](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/global-styles-and-settings.php#L108) to core. I suggest we revisit that code once this conversation settles.

We've tried to avoid classic themes to have any unwanted styles so far. If we go ahead with this, every time we add something to the default `theme.json` it'll be shipped to classic themes. I have strong reservations against doing this because it's a source of bugs for classic themes (nobody will check how a change in the default `theme.json` will affect them in the future).